packages/components/src/internal/components/samples/SampleCreationTypeModal.tsx

```diff
@@ -0,0 +1,144 @@
+            numPerParent: 1,
+            submitting: false,
+        }
+    }
+
+    onCancel = () => {
+        this.props.onCancel();
+    };
+
+    onChange = event => {
+        const { name, value } = event.target;
+        this.setState({ [name]: value } as State);
+    };
+
+    renderNumPerParent() : React.ReactNode {
+        const { parentCount, options } = this.props;
+        const { creationType, numPerParent } = this.state;
+
+        const selectedOption = options.find(option => option.type === creationType);
+        return (
+            <>
+                {this.shouldDisplayOptions() && <hr/>}
+                <div>
+                    <label className="creation-type-modal-label">{selectedOption.quantityLabel}</label>
+                    <label className="creation-type-modal-label">
+                        <FormControl
+                            className="creation-per-parent-select"
+                            min={1}
+                            max={MAX_EDITABLE_GRID_ROWS/parentCount}
+                            step={1}
+                            name={"numPerParent"}
+                            onChange={this.onChange}
+                            type="number"
+                            value={numPerParent}
+                        />
+                    </label>
+                </div>
+                Sample details and quantities can be modified on the grid.
+            </>
+        )
+    }
+
+    onConfirm = () => {
+        this.props.onSubmit(this.state.creationType, this.state.numPerParent);
+    }
+
+    shouldDisplayOptions() : boolean {
+        return this.getOptionsToDisplay().length > 1;
+    }
+
+    shouldRenderOption(option: SampleCreationTypeModel) : boolean {
+        return this.props.parentCount >= option.minParentsPerSample;
+    }
+
+    getOptionsToDisplay() : Array<SampleCreationTypeModel> {
+        return this.props.options.filter(option => this.shouldRenderOption(option));
+    }
+
+    renderOptions() : Array<React.ReactNode> {
+        const { showIcons, options } = this.props;
+        const displayOptions = this.getOptionsToDisplay();
+        if (displayOptions.length < 2)
+            return null;
+
+        let optionSet = [];
+        options.forEach((option, i) => {
+            if (this.shouldRenderOption(option)) {
+                optionSet.push(
+                    <SampleCreationTypeOption
+                        key={i}
+                        option={option}
+                        isSelected={this.state.creationType === option.type}
+                        onChoose={this.onChange}
+                        showIcon={showIcons}
+                    />
+                )
+            }
+        });
+        return optionSet;
+    }
+
+    render() : React.ReactNode {
+        const { show, parentCount } = this.props;
+        const { submitting, numPerParent } = this.state;
+
+        const parentNoun = parentCount > 1 ? 'Parents' : 'Parent';
```

Comment:
we have an api-js Utils helper for this, which I always forget about. The name has "plural" in it or "pluralize" or something like that.